DisplaySettings/DisplaySettings.h

```diff
@@ -218,24 +218,38 @@ namespace WPEFramework {
 	    bool sendHdmiCecSinkAudioDevicePowerOn();
 	    bool getHdmiCecSinkCecEnableStatus();
 	    bool getHdmiCecSinkAudioDeviceConnectedStatus();
-	    static void  cecArcRoutingThread();
+	    void sendAudioDevicePowerOnMsg();
+	    void sendArcInitiationRequestMsg();
+	    void sendArcTerminationRequestMsg();
+	    void sendAudioDevicePowerStatusRequestMsg();
+	    bool sendShortAudioDescriptorRequestMsg();
+
 	    void onTimer();
 	    void stopCecTimeAndUnsubscribeEvent();
             void checkAudioDeviceDetectionTimer();
+	    void checkArcDeviceConnected();
+	    void checkSADUpdate();
+	    void checkAudioDevicePowerStatusTimer();
 
 	    TpTimer m_timer;
             TpTimer m_AudioDeviceDetectTimer;
+	    TpTimer m_SADDetectionTimer;
+	    TpTimer m_ArcDetectionTimer;
+	    TpTimer m_AudioDevicePowerOnStatusTimer;
             bool m_subscribed;
             std::mutex m_callMutex;
+            std::mutex m_SadMutex;
 	    std::thread m_arcRoutingThread;
 	    std::mutex m_arcRoutingStateMutex;
+	    std::mutex m_ArcDisableMutex;
 	    bool m_cecArcRoutingThreadRun; 
 	    std::condition_variable arcRoutingCV;
 	    bool m_hdmiInAudioDeviceConnected;
-        bool m_arcAudioEnabled;
+            bool m_arcEarcAudioEnabled;
             bool m_arcPendingSADRequest;   
```

Comment:
was there a though to reduce the member variable as per reacting

Reply:
We are in very much required of these 3 variables as of now.